pkgs/by-name/ca/catppuccin-papirus-folders/package.nix

```diff
@@ -31,27 +33,43 @@ let
     "mocha"
   ];
   pname = "catppuccin-papirus-folders";
+
+  # Note(sewer56):
+  # Fetch the papirus-folders script from upstream
+  # Per instructions in the papirus-folders project.
+  # We fetch specific latest tested commit to ensure reproducibility.
+  papirus-folders-script = fetchurl {
+    url = "https://raw.githubusercontent.com/PapirusDevelopmentTeam/papirus-folders/0f838ee5679229e3a3e97e3b333c222c9e9615b4/papirus-folders";
+    sha256 = "sha256-swpoSKAGkDAqzP/AUFSSGLCxFNMXiyi9OhaJGBeCGwY=";
+  };
 in
 lib.checkListOfEnum "${pname}: accent colors" validAccents [ accent ] lib.checkListOfEnum
   "${pname}: flavors"
   validFlavors
   [ flavor ]
-
   stdenvNoCC.mkDerivation
   {
     inherit pname;
-    version = "unstable-2023-08-02";
+    version = "unstable-2024-08-06";
 
     src = fetchFromGitHub {
       owner = "catppuccin";
       repo = "papirus-folders";
-      rev = "fcf96737fffc343a1bf129732c37d19f2d77fa5c";
-      sha256 = "sha256-3yjIGzN+/moP2OVGDIAy4zPqUroSjx3c2mJjdZGhTsY=";
+      rev = "f83671d17ea67e335b34f8028a7e6d78bca735d7";
+      sha256 = "sha256-FiZdwzsaMhS+5EYTcVU1LVax2H1FidQw97xZklNH2R4=";
     };
 
-    nativeBuildInputs = [ gtk3 ];
+    # This gets stuck on fixup for some reason. Please help.
+    dontFixup = true;
+    nativeBuildInputs = [
+      gtk3
+      getent
+    ];
 
     postPatch = ''
+      # Copy the upstream papirus-folders script (pinned), per papirus-folders documentation
+      cp ${papirus-folders-script} ./papirus-folders
+      chmod +x ./papirus-folders
       patchShebangs ./papirus-folders
```

Comment:
Likewise, this can be done within `fetchurl`'s `postFetch` script.

Although, I'm guessing a copy or symlink is still needed if this script is used later in the build. A symlink would be preferred, if possible.

Reply:
It was needed, unfortunately I couldn't call `patchShebangs` from postFetch.
I was able to clean up a bit by fetchUrl with executable, thanks.

Reply:
It should be possible if we also add whatever hook provides `patchShebangs` to fetchurl's `nativeBuildInputs`. Not strictly needed, though.

Reply:
Ah, I see:

```
error: fixed-output derivations must not reference store paths: '/nix/store/998spbf7ifzzc4iwi6i4i0132dcyh8hd-papirus-folders.drv' references 1 distinct paths, e.g. '/nix/store/gkwbw9nzbkbz298njbn3577zmrnglbbi-bash-5.3p0'
```

A fixed-output derivation cannot depend on a _non_-fixed-output derivation, in this case bash.

Makes sense.